### ISSUE TITLE
Cannot connect from Windows 10 to Manjaro

### ISSUE THREAD

**Issue by Richardk2n** [FreeRDP/FreeRDP#7689]
Running `freerdp-shadow-cli` on my manjaro PC prooduces the following output when trying to connect from a windows 10 PC via the remote desktop gui:

```
[16:57:00:957] [23109:23109] [INFO][com.freerdp.server.shadow.x11] - X11 Extensions: XFixes: 1 Xinerama: 1 XDamage: 0 XShm: 0
[16:57:00:968] [23109:23109] [INFO][com.freerdp.core.listener] - Listening on [0.0.0.0]:3389
[16:57:06:404] [23109:23116] [DEBUG][com.winpr.thread] - Thread running, setting to detached state!
[16:57:06:419] [23109:23143] [DEBUG][com.freerdp.core.nego] - received cookie [Cookie: mstshash=richard]
[16:57:06:419] [23109:23143] [DEBUG][com.freerdp.core.nego] - RDP_NEG_REQ: RequestedProtocol: 0x0000000B
[16:57:06:420] [23109:23143] [INFO][com.freerdp.core.connection] - Client Security: NLA:1 TLS:1 RDP:0
[16:57:06:420] [23109:23143] [INFO][com.freerdp.core.connection] - Server Security: NLA:0 TLS:1 RDP:1
[16:57:06:420] [23109:23143] [INFO][com.freerdp.core.connection] - Negotiated Security: NLA:0 TLS:1 RDP:0
[16:57:06:444] [23109:23143] [ERROR][com.freerdp.core.transport] - BIO_read returned a system error 104: Connection reset by peer
[16:57:06:444] [23109:23143] [ERROR][com.freerdp.core] - transport_read_layer:freerdp_set_last_error_ex ERRCONNECT_CONNECT_TRANSPORT_FAILED [0x0002000D]
[16:57:06:444] [23109:23143] [DEBUG][com.freerdp.core.transport] - transport_check_fds: transport_read_pdu() - -1
[16:57:06:444] [23109:23143] [DEBUG][com.freerdp.core.rdp] - transport_check_fds() - -1
[16:57:06:444] [23109:23143] [ERROR][com.freerdp.client.shadow] - Failed to check FreeRDP file descriptor
[16:57:09:689] [23109:23116] [DEBUG][com.winpr.thread] - Thread running, setting to detached state!
[16:57:09:695] [23109:23153] [DEBUG][com.freerdp.core.nego] - received cookie [Cookie: mstshash=richard]
[16:57:09:695] [23109:23153] [DEBUG][com.freerdp.core.nego] - RDP_NEG_REQ: RequestedProtocol: 0x0000000B
[16:57:09:695] [23109:23153] [INFO][com.freerdp.core.connection] - Client Security: NLA:1 TLS:1 RDP:0
[16:57:09:695] [23109:23153] [INFO][com.freerdp.core.connection] - Server Security: NLA:0 TLS:1 RDP:1
[16:57:09:695] [23109:23153] [INFO][com.freerdp.core.connection] - Negotiated Security: NLA:0 TLS:1 RDP:0
[16:57:09:705] [23109:23153] [INFO][com.freerdp.core.connection] - Accepted client: RICHARD-LAPTOP
[16:57:09:705] [23109:23153] [INFO][com.freerdp.core.connection] - Accepted channels:
[16:57:09:705] [23109:23153] [INFO][com.freerdp.core.connection] -  rdpdr
[16:57:09:705] [23109:23153] [INFO][com.freerdp.core.connection] -  rdpsnd
[16:57:09:705] [23109:23153] [INFO][com.freerdp.core.connection] -  cliprdr
[16:57:09:705] [23109:23153] [INFO][com.freerdp.core.connection] -  drdynvc
[16:57:09:705] [23109:23153] [INFO][com.freerdp.core.gcc] - Active rdp encryption level: NONE
[16:57:09:705] [23109:23153] [INFO][com.freerdp.core.gcc] - Selected rdp encryption method: NONE
[16:57:09:736] [23109:23153] [DEBUG][com.freerdp.core.peer] - Received PDU_TYPE_CONFIRM_ACTIVE
[16:57:09:736] [23109:23153] [DEBUG][com.freerdp.core.rdp] - rdp_send_data_pdu: sending data (type=0x1f size=37 channelId=1009)
[16:57:09:736] [23109:23153] [DEBUG][com.freerdp.core.rdp] - rdp_send_data_pdu: sending data (type=0x14 size=41 channelId=1009)
[16:57:09:736] [23109:23153] [DEBUG][com.freerdp.core.peer] - Received PDU_TYPE_DATA
[16:57:09:736] [23109:23153] [DEBUG][com.freerdp.core.peer] - Received PDU_TYPE_DATA
[16:57:09:736] [23109:23153] [DEBUG][com.freerdp.core.peer] - Received PDU_TYPE_DATA
[16:57:09:736] [23109:23153] [DEBUG][com.freerdp.core.rdp] - rdp_send_data_pdu: sending data (type=0x14 size=41 channelId=1009)
[16:57:09:740] [23109:23153] [WARN][com.freerdp.server.shadow.x11] - x11_shadow_input_synchronize_event not implemented
[16:57:09:740] [23109:23153] [DEBUG][com.freerdp.core.peer] - Received PDU_TYPE_DATA
[16:57:09:740] [23109:23153] [DEBUG][com.freerdp.core.rdp] - rdp_send_data_pdu: sending data (type=0x28 size=41 channelId=1009)
[16:57:09:740] [23109:23153] [INFO][com.freerdp.client.shadow] - Client from 192.168.188.29 is activated (2560x1440@32)
[16:57:09:805] [23109:23153] [ERROR][com.freerdp.server.shadow.x11] - pam_authenticate failure: Conversation error
[16:57:09:805] [23109:23153] [ERROR][com.freerdp.client.shadow] - client authentication failure: -1
[16:57:09:805] [23109:23153] [ERROR][com.freerdp.core.peer] - peer_recv_callback: CONNECTION_STATE_ACTIVE - peer_recv_pdu() fail
[16:57:09:805] [23109:23153] [ERROR][com.freerdp.core.transport] - transport_check_fds: transport->ReceiveCallback() - -1
[16:57:09:805] [23109:23153] [DEBUG][com.freerdp.core.rdp] - transport_check_fds() - -1
[16:57:09:805] [23109:23153] [ERROR][com.freerdp.client.shadow] - Failed to check FreeRDP file descriptor
```

On the Windows machine a black window appears and after a very short time a error popup appears claiming a network issue.

The build configuration is as follows
```
[17:00:53:754] [24157:24157] [INFO][com.freerdp.server.shadow] - Build configuration: BUILD_TESTING=OFF BUILTIN_CHANNELS=ON HAVE_AIO_H=1 HAVE_EXECINFO_H=1 HAVE_FCNTL_H=1 HAVE_GETLOGIN_R=1 HAVE_GETPWUID_R=1 HAVE_INTTYPES_H=1 HAVE_JOURNALD_H=TRUE HAVE_MATH_C99_LONG_DOUBLE=1 HAVE_POLL_H=1 HAVE_PTHREAD_MUTEX_TIMEDLOCK=ON HAVE_PTHREAD_MUTEX_TIMEDLOCK_LIBS= HAVE_PTHREAD_MUTEX_TIMEDLOCK_SYMBOL=1 HAVE_SYSLOG_H=1 HAVE_SYS_EVENTFD_H=1 HAVE_SYS_FILIO_H= HAVE_SYS_MODEM_H= HAVE_SYS_SELECT_H=1 HAVE_SYS_SOCKIO_H= HAVE_SYS_STRTIO_H= HAVE_SYS_TIMERFD_H=1 HAVE_TM_GMTOFF=1 HAVE_UNISTD_H=1 HAVE_XI_TOUCH_CLASS=1 WITH_ALSA=ON WITH_CAIRO=OFF WITH_CCACHE=ON WITH_CHANNELS=ON WITH_CLANG_FORMAT=ON WITH_CLIENT=ON WITH_CLIENT_AVAILABLE=1 WITH_CLIENT_CHANNELS=ON WITH_CLIENT_CHANNELS_AVAILABLE=1 WITH_CLIENT_COMMON=ON WITH_CLIENT_INTERFACE=OFF WITH_CUPS=ON WITH_DEBUG_ALL=OFF WITH_DEBUG_CAPABILITIES=OFF WITH_DEBUG_CERTIFICATE=OFF WITH_DEBUG_CHANNELS=OFF WITH_DEBUG_CLIPRDR=OFF WITH_DEBUG_DVC=OFF WITH_DEBUG_KBD=OFF WITH_DEBUG_LICENSE=OFF WITH_DEBUG_MUTEX=OFF WITH_DEBUG_NEGO=OFF WITH_DEBUG_NLA=OFF WITH_DEBUG_NTLM=OFF WITH_DEBUG_RAIL=OFF WITH_DEBUG_RDP=OFF WITH_DEBUG_RDPDR=OFF WITH_DEBUG_RDPEI=OFF WITH_DEBUG_RDPGFX=OFF WITH_DEBUG_REDIR=OFF WITH_DEBUG_RFX=OFF WITH_DEBUG_RINGBUFFER=OFF WITH_DEBUG_SCARD=OFF WITH_DEBUG_SND=OFF WITH_DEBUG_SVC=OFF WITH_DEBUG_SYMBOLS=OFF WITH_DEBUG_THREADS=OFF WITH_DEBUG_TIMEZONE=OFF WITH_DEBUG_TRANSPORT=OFF WITH_DEBUG_TSG=OFF WITH_DEBUG_TSMF=OFF WITH_DEBUG_TSMF=OFF WITH_DEBUG_TSMF_AVAILABLE=0 WITH_DEBUG_URBDRC=OFF WITH_DEBUG_WND=OFF WITH_DEBUG_X11=OFF WITH_DEBUG_X11_CLIPRDR=OFF WITH_DEBUG_X11_LOCAL_MOVESIZE=OFF WITH_DEBUG_XV=OFF WITH_DSP_EXPERIMENTAL=OFF WITH_DSP_FFMPEG=ON WITH_EVENTFD_READ_WRITE=1 WITH_FAAC=OFF WITH_FAAD2=OFF WITH_FFMPEG=TRUE WITH_FFMPEG=TRUE WITH_GFX_H264=ON WITH_GPROF=OFF WITH_GSM=OFF WITH_GSSAPI=OFF WITH_ICU=ON WITH_IPP=OFF WITH_JPEG=ON WITH_LAME=OFF WITH_LIBRARY_VERSIONING=ON WITH_LIBSYSTEMD=ON WITH_MACAUDIO=OFF WITH_MACAUDIO=OFF WITH_MACAUDIO_AVAILABLE=0 WITH_MANPAGES=ON WITH_MBEDTLS=OFF WITH_OPENCL=OFF WITH_OPENH264=OFF WITH_OPENSLES=OFF WITH_OPENSSL=ON WITH_OSS=ON WITH_PAM=ON WITH_PCSC=ON WITH_PROFILER=OFF WITH_PROXY=ON WITH_PROXY_MODULES=OFF WITH_PULSE=ON WITH_SAMPLE=OFF WITH_SANITIZE_ADDRESS=OFF WITH_SANITIZE_ADDRESS_AVAILABLE=1 WITH_SANITIZE_MEMORY=OFF WITH_SANITIZE_MEMORY_AVAILABLE=1 WITH_SANITIZE_THREAD=OFF WITH_SANITIZE_THREAD_AVAILABLE=1 WITH_SERVER=ON WITH_SERVER_CHANNELS=ON WITH_SERVER_INTERFACE=ON WITH_SHADOW=ON WITH_SMARTCARD_INSPECT=OFF WITH_SOXR=OFF WITH_SSE2=ON WITH_SWSCALE=ON WITH_THIRD_PARTY=OFF WITH_VAAPI=OFF WITH_VALGRIND_MEMCHECK=OFF WITH_VALGRIND_MEMCHECK_AVAILABLE=1 WITH_WAYLAND=ON WITH_WINPR_TOOLS=ON WITH_X11=ON WITH_XCURSOR=ON WITH_XDAMAGE=ON WITH_XEXT=ON WITH_XFIXES=ON WITH_XI=ON WITH_XINERAMA=ON WITH_XKBFILE=ON WITH_XRANDR=ON WITH_XRENDER=ON WITH_XSHM=ON WITH_XTEST=ON WITH_XV=ON WITH_ZLIB=ON
Build type:          None
CFLAGS:              -march=x86-64 -mtune=generic -O2 -pipe -fno-plt -fexceptions         -Wp,-D_FORTIFY_SOURCE=2 -Wformat -Werror=format-security         -fstack-clash-protection -fcf-protection -flto -fPIC -Wall -Wno-unused-result -Wno-unused-but-set-variable -Wno-deprecated-declarations -fvisibility=hidden -Wimplicit-function-declaration -Wredundant-decls -g -fno-omit-frame-pointer -DWINPR_DLL
Compiler:            GNU, 11.2.0
Target architecture: x64
```
There is never any password prompt.
In case it is important:
The Manjaro PC has multiple monitors while the Windows PC only has one (with a higher resolution however)

**Comment by akallabeth** [comment]
Try to start with `-sec-nla` or (dangerous) `-auth`, `NLA` requires additional setup as `PAM` requires your password to compare and `NLA` uses `NTLM` hashes which do not work for that.

**Comment by Richardk2n** [comment]
sorry I actually made I mistake

The output foor `-sec-nla` is the following:
```
[17:35:15:315] [35548:35548] [INFO][com.freerdp.server.shadow.x11] - X11 Extensions: XFixes: 1 Xinerama: 1 XDamage: 0 XShm: 0
[17:35:15:319] [35548:35548] [INFO][com.freerdp.core.listener] - Listening on [0.0.0.0]:3389
[17:35:18:729] [35548:35550] [DEBUG][com.winpr.thread] - Thread running, setting to detached state!
[17:35:18:731] [35548:35562] [DEBUG][com.freerdp.core.nego] - RDP_NEG_REQ: RequestedProtocol: 0x0000000B
[17:35:18:731] [35548:35562] [INFO][com.freerdp.core.connection] - Client Security: NLA:1 TLS:1 RDP:0
[17:35:18:731] [35548:35562] [INFO][com.freerdp.core.connection] - Server Security: NLA:0 TLS:1 RDP:1
[17:35:18:731] [35548:35562] [INFO][com.freerdp.core.connection] - Negotiated Security: NLA:0 TLS:1 RDP:0
[17:35:18:740] [35548:35562] [ERROR][com.freerdp.core.transport] - BIO_read returned a system error 104: Connection reset by peer
[17:35:18:740] [35548:35562] [ERROR][com.freerdp.core] - transport_read_layer:freerdp_set_last_error_ex ERRCONNECT_CONNECT_TRANSPORT_FAILED [0x0002000D]
[17:35:18:740] [35548:35562] [DEBUG][com.freerdp.core.transport] - transport_check_fds: transport_read_pdu() - -1
[17:35:18:740] [35548:35562] [DEBUG][com.freerdp.core.rdp] - transport_check_fds() - -1
[17:35:18:740] [35548:35562] [ERROR][com.freerdp.client.shadow] - Failed to check FreeRDP file descriptor
[17:35:19:991] [35548:35550] [DEBUG][com.winpr.thread] - Thread running, setting to detached state!
[17:35:19:993] [35548:35569] [DEBUG][com.freerdp.core.nego] - RDP_NEG_REQ: RequestedProtocol: 0x0000000B
[17:35:19:993] [35548:35569] [INFO][com.freerdp.core.connection] - Client Security: NLA:1 TLS:1 RDP:0
[17:35:19:993] [35548:35569] [INFO][com.freerdp.core.connection] - Server Security: NLA:0 TLS:1 RDP:1
[17:35:19:993] [35548:35569] [INFO][com.freerdp.core.connection] - Negotiated Security: NLA:0 TLS:1 RDP:0
[17:35:19:004] [35548:35569] [INFO][com.freerdp.core.connection] - Accepted client: RICHARD-LAPTOP
[17:35:19:004] [35548:35569] [INFO][com.freerdp.core.connection] - Accepted channels:
[17:35:19:004] [35548:35569] [INFO][com.freerdp.core.connection] -  rdpdr
[17:35:19:004] [35548:35569] [INFO][com.freerdp.core.connection] -  rdpsnd
[17:35:19:004] [35548:35569] [INFO][com.freerdp.core.connection] -  cliprdr
[17:35:19:004] [35548:35569] [INFO][com.freerdp.core.connection] -  drdynvc
[17:35:19:004] [35548:35569] [INFO][com.freerdp.core.gcc] - Active rdp encryption level: NONE
[17:35:19:004] [35548:35569] [INFO][com.freerdp.core.gcc] - Selected rdp encryption method: NONE
[17:35:19:035] [35548:35569] [DEBUG][com.freerdp.core.peer] - Received PDU_TYPE_CONFIRM_ACTIVE
[17:35:19:035] [35548:35569] [DEBUG][com.freerdp.core.rdp] - rdp_send_data_pdu: sending data (type=0x1f size=37 channelId=1009)
[17:35:19:035] [35548:35569] [DEBUG][com.freerdp.core.rdp] - rdp_send_data_pdu: sending data (type=0x14 size=41 channelId=1009)
[17:35:19:035] [35548:35569] [DEBUG][com.freerdp.core.peer] - Received PDU_TYPE_DATA
[17:35:19:035] [35548:35569] [DEBUG][com.freerdp.core.peer] - Received PDU_TYPE_DATA
[17:35:19:035] [35548:35569] [DEBUG][com.freerdp.core.peer] - Received PDU_TYPE_DATA
[17:35:19:035] [35548:35569] [DEBUG][com.freerdp.core.rdp] - rdp_send_data_pdu: sending data (type=0x14 size=41 channelId=1009)
[17:35:19:035] [35548:35569] [WARN][com.freerdp.server.shadow.x11] - x11_shadow_input_synchronize_event not implemented
[17:35:19:035] [35548:35569] [DEBUG][com.freerdp.core.peer] - Received PDU_TYPE_DATA
[17:35:19:035] [35548:35569] [DEBUG][com.freerdp.core.rdp] - rdp_send_data_pdu: sending data (type=0x28 size=41 channelId=1009)
[17:35:19:035] [35548:35569] [INFO][com.freerdp.client.shadow] - Client from 192.168.188.29 is activated (2560x1440@32)
[17:35:19:039] [35548:35569] [ERROR][com.freerdp.server.shadow.x11] - pam_authenticate failure: Conversation error
[17:35:19:039] [35548:35569] [ERROR][com.freerdp.client.shadow] - client authentication failure: -1
[17:35:19:039] [35548:35569] [ERROR][com.freerdp.core.peer] - peer_recv_callback: CONNECTION_STATE_ACTIVE - peer_recv_pdu() fail
[17:35:19:039] [35548:35569] [ERROR][com.freerdp.core.transport] - transport_check_fds: transport->ReceiveCallback() - -1
[17:35:19:039] [35548:35569] [DEBUG][com.freerdp.core.rdp] - transport_check_fds() - -1
[17:35:19:039] [35548:35569] [ERROR][com.freerdp.client.shadow] - Failed to check FreeRDP file descriptor
```
And it still throws the network error.

`-auth` works

**Comment by akallabeth** [comment]
@Richardk2n ok, I see. This might be an issue as `PAM` is restricted to certain groups which normal users are not members of.
Before you go on, know that `shadow-server` is a small utility that might not fit your needs (it is hard to configure right and has quite rough edges).

So, if you just need a RDP server for linux, look at `xrdp` or `ogon-project` or check `gnome-remote-desktop` which allows similar stuff.

**Comment by Richardk2n** [comment]
Well I made a major error.
i did not realize, that `-` is not a general option sign here, but removes options.

I got it working now with nla, thank you for your time.

I do just need an rdp server, I only ever knew about xrdp, but that seems to not work with KDE5.
Thank you for pointing out the other ones. I just today found this. The setup required was more than xrdp would be, but not that much, a little documentation could turn this into a very valuable tool.
